logstash-core/lib/logstash/api/lib/app/service.rb

```diff
@@ -23,6 +23,10 @@ def agent
     LogStash::Instrument::Collector.instance.agent
   end
 
+  def started?
+    !@snapshot.nil? && has_counters?
```

Comment:
basically yes, this end up here I guess as legacy :-P

On Fri, Mar 11, 2016 at 4:44 PM João Duarte <notifications@github.com>
wrote:

> In logstash-core/lib/logstash/api/lib/app/service.rb
> <https://github.com/elastic/logstash/pull/4782#discussion_r55845737>:
>
> > @@ -23,6 +23,10 @@ def agent
> >      LogStash::Instrument::Collector.instance.agent
> >    end
> >
> > +  def started?
> > +    !@snapshot.nil? && has_counters?
>
> since we check if @snapshot <https://github.com/snapshot> is nil in
> has_counters?, isn't this first check redundant?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/logstash/pull/4782/files#r55845737>.
>